test/e2e/baremetal_test.go

```diff
@@ -25,27 +25,20 @@ func TestBaremetalDefaults(t *testing.T) {
 	defer framework.MustRemoveImageRegistry(t, client)
 	framework.MustDeployImageRegistry(t, client, nil)
 	cr := framework.MustEnsureImageRegistryIsProcessed(t, client)
-	conds := framework.GetImageRegistryConditions(cr)
-	if !conds.Degraded.IsTrue() {
-		t.Errorf("the operator is expected to be degraded, got: %s", conds)
-	}
-	if want := "StorageNotConfigured"; conds.Degraded.Reason() != want {
-		t.Errorf("degraded reason: got %q, want %q", conds.Degraded.Reason(), want)
-	}
 
 	clusterOperator := framework.MustEnsureClusterOperatorStatusIsSet(t, client)
 	for _, cond := range clusterOperator.Status.Conditions {
 		switch cond.Type {
 		case configapiv1.OperatorAvailable:
-			if cond.Status != configapiv1.ConditionFalse {
-				t.Errorf("expected clusteroperator to report Available=%s, got %s", configapiv1.ConditionFalse, cond.Status)
-			}
-		case configapiv1.OperatorDegraded:
 			if cond.Status != configapiv1.ConditionTrue {
-				t.Errorf("expected clusteroperator to report Degraded=%s, got %s", configapiv1.ConditionTrue, cond.Status)
+				t.Errorf("expected clusteroperator to report Available=%s, got %s", configapiv1.ConditionTrue, cond.Status)
+			}
+			if cond.Reason != "Removed" {
+				t.Errorf("expected reason to be 'Removed', got %s", cond.Reason)
```

Comment:
@ricardomaraschini I believe that we have a utility method to ensure that the cluster operator is reporting itself "green" (available=true, progressing=false, degraded=false) - `MustEnsureClusterOperatorStatusIsNormal`

Reply:
Good one! I also think it is important to check if `Reason` is properly set to "Removed" in all `Conditions`. I am building another solution using another already provided function called `framework.ConditionExistsWithStatusAndReason`. PTAL.